lib/command/src/Obelisk/Command/Thunk.hs

```diff
@@ -693,12 +727,15 @@ packThunk :: MonadObelisk m => ThunkPackConfig -> FilePath -> m ThunkPtr
 packThunk = packThunk' False
 
 packThunk' :: MonadObelisk m => Bool -> ThunkPackConfig -> FilePath -> m ThunkPtr
-packThunk' noTrail (ThunkPackConfig force thunkConfig) thunkDir = checkThunkDirectory "Can't pack/unpack from within the thunk directory" thunkDir >> readThunk thunkDir >>= \case
+packThunk' noTrail (ThunkPackConfig force thunkConfig tScheme) thunkDir = checkThunkDirectory "Can't pack/unpack from within the thunk directory" thunkDir >> readThunk thunkDir >>= \case
   Left err -> failWith $ T.pack $ "thunk pack: " <> show err
   Right (ThunkData_Packed _) -> failWith "pack: thunk is already packed"
   Right (ThunkData_Checkout _) -> do
     withSpinner' ("Packing thunk " <> T.pack thunkDir)
                  (finalMsg noTrail $ const $ "Packed thunk " <> T.pack thunkDir) $ do
+      let overrideScheme = applyThunkScheme tScheme
+      let overrideSchemeThunkPtr = thunkPtr_source . _ThunkSource_Git %~ overrideScheme
+      thunkPtr <- overrideSchemeThunkPtr <$> getThunkPtr (not force) thunkDir
```

Comment:
This "feigned" mutation via shadowing isn't common in Haskell (it is in OCaml!) so let's just try to put this all in the same line.

Reply:
Would it make sense to move this logic into `modifyThunkPtrByConfig`?

Reply:
Looking back on this. I think the scheme-overridden ptr wasn't used :open_mouth:

Reply:
I am confused: What shadowing?